pkg/gardenlet/controller/backupbucket/backup_bucket_reconciler.go

```diff
@@ -132,20 +140,40 @@ func (r *reconciler) reconcileBackupBucket(ctx context.Context, gardenClient kub
 		r.recorder.Eventf(backupBucket, corev1.EventTypeWarning, gardencorev1beta1.EventReconcileError, "%s", reconcileErr.Description)
 
 		if updateErr := updateBackupBucketStatusError(ctx, gardenClient.DirectClient(), backupBucket, reconcileErr.Description+" Operation will be retried.", reconcileErr); updateErr != nil {
-			backupBucketLogger.Errorf("Could not update the BackupBucket status after deletion error: %+v", updateErr)
-			return reconcile.Result{}, updateErr
+			backupBucketLogger.Errorf("Could not update the BackupBucket status: %+v", updateErr)
+		}
+
+		if updateErr := patchSeedCondition(ctx, gardenClient.Client(), seed, gardencorev1beta1helper.UpdatedCondition(conditionBackupBucketReady,
+			gardencorev1beta1.ConditionFalse, "BackupBucketReconciliationError", reconcileErr.Description)); updateErr != nil {
+			backupBucketLogger.Errorf("Could not update Seed status: %+v", updateErr)
 		}
+
 		return reconcile.Result{}, errors.New(reconcileErr.Description)
 	}
 
-	if updateErr := updateBackupBucketStatusSucceeded(ctx, gardenClient.DirectClient(), backupBucket, "Backup Bucket has been successfully reconciled."); updateErr != nil {
+	msg := "Backup Bucket has been reconciled successfully."
+	if updateErr := updateBackupBucketStatusSucceeded(ctx, gardenClient.DirectClient(), backupBucket, msg); updateErr != nil {
 		backupBucketLogger.Errorf("Could not update the Shoot status after reconciliation success: %+v", updateErr)
 		return reconcile.Result{}, updateErr
 	}
 
+	if updateErr := patchSeedCondition(ctx, gardenClient.Client(), seed, gardencorev1beta1helper.UpdatedCondition(conditionBackupBucketReady,
+		gardencorev1beta1.ConditionTrue, "BackupBucketAvailable", msg)); updateErr != nil {
+		return reconcile.Result{}, updateErr
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func patchSeedCondition(ctx context.Context, c client.Client, seed *gardencorev1beta1.Seed, condition gardencorev1beta1.Condition) error {
```

Comment:
Though at present, there is one backup bucket per seed, technically, there could be more than one backup bucket per seed. Does it make sense to handle this case already?

Reply:
Good point @amshuman-kr, I guess it makes sense. A `BackupBucket` resource can (not today, but will hopefully be implemented in the future as part of GEP-7?, cc @plkokanov) be moved to a new seed, and this new seed will then have > 1 associated `BackupBucket`s.